homeassistant/components/livisi/switch.py

```diff
@@ -0,0 +1,170 @@
+"""Code to handle a Livisi switches."""
+from __future__ import annotations
+
+from typing import Any
+
+from homeassistant.components.switch import SwitchEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import (
+    DOMAIN,
+    LIVISI_REACHABILITY_CHANGE,
+    LIVISI_STATE_CHANGE,
+    LOGGER,
+    PSS_DEVICE_TYPE,
+)
+from .coordinator import LivisiDataUpdateCoordinator
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up switch device."""
+    coordinator: LivisiDataUpdateCoordinator = hass.data[DOMAIN][config_entry.entry_id]
+
+    @callback
+    def handle_coordinator_update() -> None:
+        """Add switch."""
+        shc_devices: list[dict[str, Any]] = coordinator.data
+        for device in shc_devices:
+            if device["type"] == PSS_DEVICE_TYPE:
+                if device not in coordinator.switch_devices:
+                    livisi_switch: SwitchEntity = create_entity(
+                        config_entry, device, coordinator
+                    )
+                    LOGGER.debug("Include device type: %s", device.get("type"))
+                    coordinator.switch_devices.append(device)
+                    async_add_entities([livisi_switch])
+
+    coordinator.async_add_listener(handle_coordinator_update)
+
+
+def create_entity(
+    config_entry: ConfigEntry,
+    device: dict[str, Any],
+    coordinator: LivisiDataUpdateCoordinator,
+) -> SwitchEntity:
+    """Create Switch Entity."""
+    config_details: dict[str, Any] = device["config"]
+    capabilities: list = device["capabilities"]
+    room_id: str = device["location"].removeprefix("/location/")
+    room_name: str = coordinator.rooms[room_id]
+    livisi_switch = LivisiSwitch(
+        config_entry,
+        coordinator,
+        unique_id=device["id"],
+        manufacturer=device["manufacturer"],
+        product=device["product"],
+        serial_number=device["serialNumber"],
+        device_type=device["type"],
+        name=config_details["name"],
+        capability_id=capabilities[0],
+        room=room_name,
+    )
+    return livisi_switch
+
+
+class LivisiSwitch(CoordinatorEntity[LivisiDataUpdateCoordinator], SwitchEntity):
+    """Represents the Livisi Switch."""
+
+    def __init__(
+        self,
+        config_entry: ConfigEntry,
+        coordinator: LivisiDataUpdateCoordinator,
+        unique_id,
```

Comment:
What's the type? Please complete the type annotations here.
```suggestion
        unique_id: str,
```

Reply:
The type is string, I added the type for annotations, thank you.